### ISSUE TITLE
UpdateFn and PostUpdate appear to be broken in Beta v3

### ISSUE THREAD

**Issue by um3k** [chrismaltby/gb-studio#567]
**Describe the bug**
UpdateFn and PostUpdate, JS UI functions used primarily by Text: Display Dialogue and Text: Display Menu to limit character counts, are no longer being called in Beta v3.

**To Reproduce**
1. Add a Text: Display Dialogue or Text: Display Menu event
2. Type something longer than the character limit, whether per line or for the entire box
3. Notice that it doesn't automatically get truncated, this is the actual bug
4. Run the game, if the character count is significantly longer than the limit, it can even overwrite background tiles

**Expected behavior**
Character limits are enforced in the UI, via the invocation of UpdateFn and PostUpdate. It also affects the plugins I have developed, which use these functions for a wider variety of things.

**Screenshots**
![image](https://user-images.githubusercontent.com/371786/94347506-de090600-0002-11eb-928d-81de645aa4bb.png)![image](https://user-images.githubusercontent.com/371786/94347484-ab5f0d80-0002-11eb-9551-cf58103df578.png)

**Platform (please complete the following information):**
 - OS: Windows 10
 - 2.0 Beta v3

**Additional context**
I initially discovered this issue when it was reported that one of my plugins had stopped working.

**Comment by RichardULZ** [comment]
Right, so the result of the issue is "dialogue and menu no longer limit characters in beta 3"
Might be an easier to understand title so we don't get duplicates of this issue.
@chrismaltby i take it this is typescript refactor related? :(

**Comment by pau-tomas** [comment]
Ah, I think https://github.com/chrismaltby/gb-studio/commit/84a836f9cf5585c804f7e2397f2f75944e88f711 caused this. Since some of the fields are functions and those aren't supported then stringifying to JSON.

**Comment by chrismaltby** [comment]
😞 yeah you're right @pau-tomas I think it will be that commit too, I'll see what I can do

**Comment by pau-tomas** [comment]
The only solution I can think is to loop thru the fields and if one is a function add it manually to the object.

**Comment by chrismaltby** [comment]
I've got a slightly different approach of keeping the events as proxy objects and only turning the default fields to a plain javascript object just before creating the update action. Just about to submit a PR 👍

**Comment by chrismaltby** [comment]
I've merged in a fix in #570 hopefully this removes the issue for you and doesn't introduce anything new
